services/src/main/java/org/exoplatform/task/util/TaskUtil.java

```diff
@@ -1295,6 +1298,27 @@ public static String getResourceBundleLabel(Locale locale, String label) {
     return resourceBundleService.getResourceBundle(resourceBundleService.getSharedResourceBundleNames(), locale).getString(label);
   }
 
+  /**
+   * Gets platform language of current user. In case of any errors return null.
+   *
+   * @return the platform language
+   */
+  public static String getCurrentUserLanguage(String userId) {
+    String lang = Locale.getDefault().getLanguage();
+    try {
+      LocaleContextInfo localeCtx = LocaleContextInfoUtils.buildLocaleContextInfo(userId);
+      LocalePolicy localePolicy = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(LocalePolicy.class);
+      if(localePolicy != null) {
+        Locale locale = localePolicy.determineLocale(localeCtx);
+        lang = locale.toString();
+      }
+      return lang;
+    } catch (Exception e) {
+      LOG.warn("Error when getting current user language" + e);
+      return lang;
+    }
+  }
+
   /**
```

Comment:
I think you could make it simplier :
```suggestion
      LocaleContextInfo localeCtx = LocaleContextInfoUtils.buildLocaleContextInfo(userId);
      LocalePolicy localePolicy = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(LocalePolicy.class);
       String lang = Locale.getDefault().getLanguage();
      if(localePolicy != null) {
        Locale locale = localePolicy.determineLocale(localeCtx);
        lang = locale.toString();
      }
      return lang;
  }

  /**
```

I dont think the try catch is necessary here